app/seeders/demo_data/project_seeder.rb

```diff
@@ -60,6 +60,13 @@ def project_content_seeder_classes
       ]
     end
 
+    def applicable?
+      types = seed_data.lookup("types")
+
+      seed_data.reference_exists?(:default_role_project_admin) &&
+        (types.nil? || types.all? { |type_reference| seed_data.reference_exists?(type_reference) })
+    end
```

Comment:
I find the `types.nil? || types.all { |ref| ... }` part a bit confusing to read.

I think it would be better to change `types = seed_data.lookup("types")` into `types = seed_data.lookup("types") || []` to get rid of the `types.nil?` part.

And even better, as the same code is used line 104 to get the types from `project_data`, a method can be extracted:

Code becomes something like this:

```suggestion
    def applicable?
      seed_data.reference_exists?(:default_role_project_admin) \
        && project_data_types.all? { |type_reference| seed_data.reference_exists?(type_reference) }
    end

    private

    def set_types
      print_status "   -Assigning types."

      project.types = seed_data.find_references(project_data_types)
    end

    def project_data_types
      project_data.lookup("types") || []
    end
```

I'm not too sure about the method name. Could also well be `type_references_data`, `types_data`, or something else.